src/main/java/teammates/common/util/LogHelper.java

```diff
@@ -0,0 +1,128 @@
+package teammates.common.util;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import teammates.common.exception.InvalidParametersException;
+
+import com.google.appengine.api.log.AppLogLine;
+import com.google.appengine.api.log.LogQuery;
+import com.google.appengine.api.log.LogServiceFactory;
+import com.google.appengine.api.log.RequestLogs;
+import com.google.appengine.api.log.LogService.LogLevel;
+import com.google.appengine.api.modules.ModulesService;
+import com.google.appengine.api.modules.ModulesServiceFactory;
+
+public class LogHelper {
+    /**
+     * 6 past versions to query, including the current version and its 5 preceding versions.
+     */
+    public static final int MAX_PAST_VERSIONS_TO_QUERY = 6;
+    
+    private LogQuery query;
+    
+    public LogHelper() {
+        query = LogQuery.Builder.withDefaults();
+    }
+    
+    /**
+     * Sets values for query.
+     * If versionsToQuery is null or empty, the current version with its 5 preceding versions will be used instead.
+     * 
+     * @param versionsToQuery 
+     * @param startTime
+     * @param endTime
+     * @param offset
+     */
+    public void setQuery(List<String> versionsToQuery, Long startTime, Long endTime, String offset) {
+        query.includeAppLogs(true);
+        query.batchSize(1000);
+        query.minLogLevel(LogLevel.INFO);
```

Comment:
Consider making `1000` and `LogLevel.INFO` a private constant, along with explanation of what they are. Perhaps the `true` in `includeAppLogs` also.